### PR TITLE
Animal traps fixes and improvement

### LINKED ISSUE #6150: Dismanting traps deletes their contents and traps do not visually display contents
Animal traps show themselves as empty. Had to LOOC some people so they knew I accidentally trapped myself in one because you can't see what's inside them visually as the sprite is shown empty, unless you examine the trap itself.

### LINKED ISSUE #6158: Telsa ball can't spell.
"All 8 energize for a second, sending their energy to the main ball, which redirects it at the nearest object! Sacraficing one of its miniballs!"

"Sacrificing" is the correct term here.

### PR DESCRIPTION
**Pull request by Sindorman**
- Fixes #6150 

- Animal traps now visually display what prey is inside of them by adding mob's icon as underlay.

- Tesla no longer logs spawn of new miniballs

- Fixes main changelog where it is supposed to be addition.

- Fixes #6158 

- Examining animal trap with prey now also displays examine message of the prey itself too.

### CODE REVIEW

File: code/game/objects/items/weapons/traps.dm [hunk 1]
```diff
@@ -152,6 +152,22 @@
...
 
+/obj/item/weapon/trap/animal/update_icon()
+	icon = initial(icon)
+	icon_state = "[icon_base][deployed]"
+	var/datum/L = captured ? captured.resolve() : null
+	if (!L)
+		deployed = FALSE
+		captured = null
+		release_time = world.time
+		update_icon()
```
Comment (Arrow768) [inline code/game/objects/items/weapons/traps.dm]: Isnt that a loop?

Comment (Sindorman) [reply code/game/objects/items/weapons/traps.dm]: Oh RIP yes. Infinite

File: code/modules/power/singularity/singularity.dm [hunk 2]
```diff
@@ -30,9 +30,10 @@
 
 	var/chained = 0//Adminbus chain-grab
 
-/obj/singularity/New(loc, var/starting_energy = 50, var/temp = 0)
+/obj/singularity/New(loc, var/starting_energy = 50, var/temp = 0, var/log = FALSE)
 	//CARN: admin-alert for chuckle-fuckery.
-	admin_investigate_setup()
+	if(log)
```
Comment (skull132) [inline code/modules/power/singularity/singularity.dm]: Would argue against this. Bad OOP and allows for weird slip-ups.

Override `/obj/singularity/proc/admin_investigate_setup` for miniballs instead.

File: code/game/objects/items/weapons/traps.dm [hunk 3]
```diff
@@ -152,6 +152,21 @@
...
+	icon_state = "[icon_base][deployed]"
+	var/datum/L = captured ? captured.resolve() : null
+	if (!L)
+		deployed = FALSE
+		captured = null
+		release_time = world.time
+		return
+	if(isliving(L))
+		var/mob/living/ll = L
+		var/image/temp = ll.appearance
```
Comment (Lohikar) [inline code/game/objects/items/weapons/traps.dm]: `appearance` is not an image.

File: code/game/objects/items/weapons/traps.dm [hunk 4]
```diff
@@ -152,6 +152,21 @@
...
+		temp.add_overlay(src.appearance)
+		icon = temp
```
Comment (Lohikar) [inline code/game/objects/items/weapons/traps.dm]: `icon` is not an image.

File: code/game/objects/items/weapons/traps.dm [hunk 5]
```diff
@@ -152,6 +152,21 @@
...
+	var/datum/L = captured ? captured.resolve() : null
+	if (!L)
+		deployed = FALSE
+		captured = null
+		release_time = world.time
+		return
+	if(isliving(L))
+		var/mob/living/ll = L
+		var/image/temp = ll.appearance
+		temp.add_overlay(src.appearance)
```
Comment (Lohikar) [inline code/game/objects/items/weapons/traps.dm]: ...does this even work? You're mutating an immutable reference.

File: code/game/objects/items/weapons/traps.dm [hunk 6]
```diff
@@ -152,21 +152,30 @@
...
 		return
 	if(isliving(L))
 		var/mutable_appearance/MA = new(L)
 		MA.layer = FLOAT_LAYER
-		MA.plane = FLOAT_PLANE
-		underlays += MA
+		MA.plane = FLOAT_PLANE	
+		var/temp = underlays.Find(MA)
+		if(temp)
+			underlays[temp] = MA
```
Comment (Lohikar) [inline code/game/objects/items/weapons/traps.dm]: ???????

File: code/game/objects/items/weapons/traps.dm [hunk 7]
```diff
@@ -349,13 +378,15 @@
...
+		STOP_PROCESSING(SSprocessing, src)
 		return
 
 	var/msg
 	if (isliving(L))
 		var/mob/living/ll = L
 		ll.forceMove(target)
 		msg = "<span class='warning'>[ll] runs out of \the [src].</span>"
+		STOP_PROCESSING(SSprocessing, src)
 	else if (istype(L, /obj/effect/spider/spiderling))
```
Comment (skull132) [inline code/game/objects/items/weapons/traps.dm]: I should not have let this code pass. I regret code.

**Pull request merged**